### PR TITLE
Pin default themes to the top for event managers

### LINKED ISSUE #3166: Request: "Default" themes pinned to the top for event administrators/managers
This concerns the instance at CERN.

When a user is not signed in or is not the administrator of an event, they are presented with a limited set of "default" themes to skin the event:

firefox - - screen shot 24 nov 2017 at 20 58 36

However, when you log in as an administrator or manager, you see a much longer list:

firefox - - screen shot 24 nov 2017 at 20 56 43

Is it possible to show this second list with the "default" themes from the first list pinned to the top and clearly separated (by a line of some sort) from the other options?

Thank you.

### PR DESCRIPTION
**Pull request by mvidalgarcia**
Resolves #3166

### CODE REVIEW

File: indico/modules/events/templates/header.html [hunk 1]
```diff
@@ -38,15 +38,20 @@
...
-            <li>
-                {% if id == theme %}
-                    <a class="disabled"><strong>{{ data.name }}</strong></a>
-                {% else %}
-                    <a href="{{ url_for(endpoint, event, view=id) }}">{{ data.name }}</a>
-                {% endif %}
-            </li>
+        {% for user_visible, themes in themes.items()|sort(attribute='1.name')|groupby('1.user_visible')|reverse
+           if event.can_manage(session.user) or user_visible %}
+            {% if user_visible == False %}
```
Comment (mic4ael) [inline indico/modules/events/templates/header.html]: either `false` or just `not user_visible`?

File: indico/modules/events/templates/header.html [hunk 2]
```diff
@@ -38,15 +38,20 @@
...
-            <li>
-                {% if id == theme %}
-                    <a class="disabled"><strong>{{ data.name }}</strong></a>
-                {% else %}
-                    <a href="{{ url_for(endpoint, event, view=id) }}">{{ data.name }}</a>
-                {% endif %}
-            </li>
+        {% for user_visible, themes in themes.items()|sort(attribute='1.name')|groupby('1.user_visible')|reverse
+           if event.can_manage(session.user) or user_visible %}
+            {% if user_visible == false %}
```
Comment (ThiefMaster) [inline indico/modules/events/templates/header.html]: `if not user_visible`

File: indico/modules/events/templates/header.html [hunk 3]
```diff
@@ -38,15 +38,20 @@
...
-           if event.can_manage(session.user) or data.user_visible %}
-            <li>
-                {% if id == theme %}
-                    <a class="disabled"><strong>{{ data.name }}</strong></a>
-                {% else %}
-                    <a href="{{ url_for(endpoint, event, view=id) }}">{{ data.name }}</a>
-                {% endif %}
-            </li>
+        {% for user_visible, themes in themes.items()|sort(attribute='1.name')|groupby('1.user_visible')|reverse
+           if event.can_manage(session.user) or user_visible %}
```
Comment (ThiefMaster) [inline indico/modules/events/templates/header.html]: It was already like this before, but I would reverse the order of the checks. Checking the truthiness of a simple variable is usually much cheaper than any method call

File: indico/modules/events/templates/header.html [hunk 4]
```diff
@@ -38,15 +38,20 @@
...
-        {% for id, data in themes.items()|sort(attribute='1.name')
-           if event.can_manage(session.user) or data.user_visible %}
-            <li>
-                {% if id == theme %}
-                    <a class="disabled"><strong>{{ data.name }}</strong></a>
-                {% else %}
-                    <a href="{{ url_for(endpoint, event, view=id) }}">{{ data.name }}</a>
-                {% endif %}
-            </li>
+        {% for user_visible, themes in themes.items()|sort(attribute='1.name')|groupby('1.user_visible')|reverse
```
Comment (ThiefMaster) [inline indico/modules/events/templates/header.html]: I think shadowing `themes` with `themes` is rather ugly. I'd try to find a different name (which might be hard) or use the grouper/list style of using groupby as shown in the docs (didn't even know that existed, just saw it when looking if there's a `reversed` attribute for groupby!): http://jinja.pocoo.org/docs/2.10/templates/#groupby

File: indico/modules/events/templates/header.html (hunk 2 above)
Comment (mvidalgarcia) [reply indico/modules/events/templates/header.html]: There are themes with `user_visible = None`

**Comment by mvidalgarcia** [comment]
Updated

File: indico/modules/events/settings.py [hunk 5]
```diff
@@ -226,6 +226,7 @@ def settings(self):
             # to override a default style so let them do so...
             for name, definition in settings.get('definitions', {}).viewitems():
                 definition['plugin'] = plugin
+                definition['user_visible'] = definition.get('user_visible', False)
```
Comment (ThiefMaster) [inline indico/modules/events/settings.py]: definition.setdefault('user_visible', False)

File: indico/modules/events/templates/header.html [hunk 6]
```diff
@@ -38,15 +38,20 @@
...
-        {% for id, data in themes.items()|sort(attribute='1.name')
-           if event.can_manage(session.user) or data.user_visible %}
-            <li>
-                {% if id == theme %}
-                    <a class="disabled"><strong>{{ data.name }}</strong></a>
-                {% else %}
-                    <a href="{{ url_for(endpoint, event, view=id) }}">{{ data.name }}</a>
-                {% endif %}
-            </li>
+        {% for grouper, list in themes.items()|sort(attribute='1.name')|groupby('1.user_visible')|reverse
```
Comment (ThiefMaster) [inline indico/modules/events/templates/header.html]: I was more thinking about `for group in ...` and then using `group.grouper` and `group.list` in the code below

Comment (ThiefMaster) [reply indico/modules/events/templates/header.html]: FWIW, if you want to keep the argument unpacking style, maybe go for `user_visible` and `themes_list`... the latter is not very nice naming but at least I can't think of anything better..

File: CHANGES.rst [hunk 7]
```diff
@@ -69,6 +69,7 @@ Improvements
   by Indico admins (:issue:`3210`)
 - Sort contribution types alphabetically
 - Add folding indicators when printing foldable badges (:issue:`3216`)
+- Pin default themes to the top for event managers (:issue:`3166`)
```
Comment (ThiefMaster) [inline CHANGES.rst]: This is in the wrong section now (should go up to the 2.1 improvements)

**Pull request merged**